cortex_xdr/icon_cortex_xdr/util/api.py

```diff
@@ -0,0 +1,302 @@
+           }
+        }
+
+        result = self._post_to_api(api_endpoint, post_body)
+        return result.get("reply")
+
+    def _isolate_endpoint(self, endpoint, isolation_state):
+        if isolation_state:
+            api_endpoint = "/public_api/v1/endpoints/isolate/"
+        else:
+            api_endpoint = "/public_api/v1/endpoints/unisolate/"
+
+        endpoint_id = endpoint[0].get("endpoint_id")
+
+        self.logger.info(f"Isolation: {isolation_state}\nEndpoint ID:{endpoint_id}")
+        post_body = {
+           "request_data":{
+              "endpoint_id": endpoint_id
+           }
+        }
+
+        result = self._post_to_api(api_endpoint, post_body)
+        return result.get("reply")
+
+    def _standard_authentication(self, api_key_id, api_key):
+        return {
+            "x-xdr-auth-id": str(api_key_id),
+            "Authorization": api_key
+        }
+
+    def _advanced_authentication(self, api_key_id: int, api_key: str) -> dict:
+        '''
+        This was generated by Cortex XDR.
+
+        This will create the headers we need to authenticate to the API
+
+        :param api_key_id: ID of the API key we are going to use, shown in the API Keys page (ex. 1, 2, 3)
+        :type api_key_id: int
+        :param api_key: The API String. This is generated when you create a new API key
+        :type api_key: string
+        :return: headers object
+        :rtype: dict
+        '''
+
+        # Generate a 64 bytes random string
+        nonce = "".join([secrets.choice(string.ascii_letters + string.digits) for _ in range(64)])
+        # Get the current timestamp as milliseconds.
+        timestamp = int(datetime.now(timezone.utc).timestamp()) * 1000
+        # Generate the auth key:
+        auth_key = "%s%s%s" % (api_key, nonce, timestamp)
+        # Convert to bytes object
+        auth_key = auth_key.encode("utf-8")
+        # Calculate sha256:
+        api_key_hash = hashlib.sha256(auth_key).hexdigest()
+        # Generate HTTP call headers
+        self.headers = {
+            "x-xdr-timestamp": str(timestamp),
+            "x-xdr-nonce": nonce,
+            "x-xdr-auth-id": str(api_key_id),
+            "Authorization": api_key_hash
+        }
+
+
+    def _post_to_api(self, endpoint, post_body):
+        url = urllib.parse.urljoin(self.fully_qualified_domain_name, endpoint)
+        try:
+            response = requests.post(url=url, json=post_body, headers=self.headers)
+
+            if response.status_code == 400:
+                resp = json.loads(response.text)
+                raise PluginException(
+                    cause=resp.get("message"),
+                    assistance="Bad request, invalid JSON.",
+                )
+            if response.status_code == 401:
+                resp = json.loads(response.text)
+                raise PluginException(
+                    cause=resp.get("message"),
+                    assistance="Authorization failed. Check your API_KEY_ID & API_KEY.",
```

Comment:
```suggestion
                    assistance="Authorization failed. Check your API_KEY_ID & API_KEY.",
```
These should match the user-facing input names